### PR TITLE
[partitioned dbs] Updates to index editor and index clone modal

### PR DESCRIPTION
**Pull request by Antonio-Maranhao**
## Overview

Seventh (and last 😀) in a series of PRs that will be submitted in support of the new user-defined partitioned databases feature (apache/couchdb#1605).

Summary:

 - Add 'Partitioned' checkbox to Index Editor
  - Checkbox is read-only for existing ddocs and is updated to reflect the selected ddoc
  - Checkbox is editable when choosing to create a new ddoc
  - CUSTOM reducer option is not shown when editing/creating partitioned views
 - Add 'Partitioned' checkbox to Clone Index modal

## Testing recommendations

On a partitioned database: 
- Click on 'New view'
   - Select 'New Document', name the new ddoc, and make sure 'Partitioned' is checked
   - Verify the 'CUSTOM' reducer is not listed
   - Click the button to create the view
   - Verify the new view is partitioned
- Use the menu to clone the view you just created
  - Select 'New Document', name the new ddoc and view, and uncheck 'Partitioned'
   - Click the button to clone the view
   - Verify the new view is global
   - Verify the original view still exists
- Click on 'New view'
   - Select 'New Document', name the new ddoc, and uncheck 'Partitioned' 
   - Verify the 'CUSTOM' reducer is listed
   - Click the button to create the view
   - Verify the new view is global

## Related Pull Requests

apache/couchdb#1605

## Checklist

- [x] Code is written and works correctly;
- [x] Changes are covered by tests;

### CODE REVIEW

**Comment by garrensmith** [comment]
When I'm creating a new view and I uncheck the partitioned button. Then add a custom reducer, then if I click the partitioned checkbox. The reducer changes to `_sum` but the custom reducer codeblock is still available.

File: app/addons/documents/index-editor/reducers.js [hunk 1]
```diff
@@ -11,11 +11,14 @@
 // the License.
 
 import ActionTypes from './actiontypes';
-import Resources from "../resources";
+import Resources from '../resources';
+import Helpers from '../helpers';
 
 const defaultMap = 'function (doc) {\n  emit(doc._id, 1);\n}';
 const defaultReduce = 'function (keys, values, rereduce) {\n  if (rereduce) {\n    return sum(values);\n  } else {\n    return values.length;\n  }\n}';
 const builtInReducers = ['_sum', '_count', '_stats'];
```
Comment (garrensmith) [inline app/addons/documents/index-editor/reducers.js]: There is the other couchdb reducer you will need to add in as well.

File: app/addons/documents/index-editor/reducers.js [hunk 2]
```diff
@@ -153,6 +179,8 @@ export default function indexEditor(state = initialState, action) {
         ...state,
         designDocId: options.value
       };
+      // newState.reduceOptions = getReduceOptions(newState);
```
Comment (garrensmith) [inline app/addons/documents/index-editor/reducers.js]: You can remove.

**Comment by Antonio-Maranhao** [comment]
@garrensmith After playing with it for a bit, I didn't like how hiding the CUSTOM option works as you flip between partitioned/global ddocs. It sort of forces the change from CUSTOM to one of the built-in reduces - which can be confusing to the user, particularly for existing views.

To make things easier, I kept the CUSTOM option always available, and the form shows a warning if it's not supported based on the ddoc selection.

**Comment by Antonio-Maranhao** [comment]
@wohali sure, will do.
The functions for partitioned dbs will only show if the couchdb server supports it - i.e. the welcome message contains the 'partitions' feature.
So it's safe to use master for the 2.3.0 release.

**Pull request merged**